File: SendMosslerTransferLetterCLG/SQLHelper.vb
Imports Microsoft.VisualBasic
Imports System.Data
Imports System.Data.SqlClient
Imports System.Configuration

Public Class SqlHelper

    Public Shared Sub ExecuteNonQuery(ByVal cmdText As String, Optional ByVal cmdType As CommandType = CommandType.StoredProcedure, Optional ByVal params() As SqlParameter = Nothing)
        Dim cmd As New SqlCommand()

        Try
            cmd.CommandType = cmdType
            cmd.CommandText = cmdText
            If Not IsNothing(params) Then
                cmd.Parameters.AddRange(params)
            End If
            cmd.Connection = New SqlConnection(ConfigurationManager.ConnectionStrings("LEXSPROD").ConnectionString)
            cmd.Connection.Open()
            cmd.ExecuteNonQuery()
        Catch ex As Exception
            Throw ex
        Finally
            If cmd.Connection.State <> ConnectionState.Closed Then
                cmd.Connection.Close()
            End If
            cmd.Dispose()
            cmd = Nothing
        End Try
    End Sub

    Public Shared Function GetDataTable(ByVal cmdText As String, Optional ByVal cmdType As CommandType = CommandType.Text, Optional ByVal params() As SqlParameter = Nothing) As DataTable

        Dim cmd As New SqlCommand()
        Dim opened As Boolean

        Try
            cmd.CommandType = cmdType
            cmd.CommandText = cmdText
            cmd.CommandTimeout = 240
            If Not IsNothing(params) Then
                cmd.Parameters.AddRange(params)
            End If
            cmd.Connection = New SqlConnection(ConfigurationManager.ConnectionStrings("LEXSPROD").ConnectionString)
            If cmd.Connection.State <> ConnectionState.Open Then
                cmd.Connection.Open()
                opened = True
            End If
            Dim dtTemp As New Data.DataTable
            dtTemp.Load(cmd.ExecuteReader())
            If opened Then
                cmd.Connection.Close()
            End If
            cmd.Dispose()
            cmd = Nothing
            Return dtTemp
        Catch ex As Exception
            Throw ex
        End Try
        'Dim cmd As New SqlCommand()
        'Dim dtTemp As New DataTable

        'Try
        '    cmd.CommandType = cmdType
        '    cmd.CommandText = cmdText
        '    cmd.CommandTimeout = 360 '6min
        '    If Not IsNothing(params) Then
        '        cmd.Parameters.AddRange(params)
        '    End If
        '    cmd.Connection = New SqlConnection(ConfigurationManager.ConnectionStrings("LEXSPROD").ConnectionString)
        '    cmd.Connection.Open()
        '    Dim reader As SqlClient.SqlDataReader
        '    reader = cmd.ExecuteReader()
        '    dtTemp.Load(reader)
        'Catch ex As Exception
        '    Throw ex
        'Finally
        '    If cmd.Connection.State <> ConnectionState.Closed Then
        '        cmd.Connection.Close()
        '    End If
        '    cmd.Dispose()
        '    cmd = Nothing
        'End Try

        'Return dtTemp
    End Function

    Public Shared Function GetDataSet(ByVal cmdText As String, Optional ByVal cmdType As CommandType = CommandType.StoredProcedure, Optional ByVal params() As SqlParameter = Nothing) As DataSet
        Dim cmd As New SqlCommand()
        Dim dsTemp As New DataSet

        Try
            cmd.CommandType = cmdType
            cmd.CommandText = cmdText
            cmd.CommandTimeout = 360 '6min
            If Not IsNothing(params) Then
                cmd.Parameters.AddRange(params)
            End If
            cmd.Connection = New SqlConnection(ConfigurationManager.ConnectionStrings("LEXSPROD").ConnectionString)
            Dim da As New SqlDataAdapter(cmd)
            da.Fill(dsTemp)
        Catch ex As Exception
            Throw ex
        Finally
            If cmd.Connection.State <> ConnectionState.Closed Then
                cmd.Connection.Close()
            End If
            cmd.Dispose()
            cmd = Nothing
        End Try

        Return dsTemp
    End Function

    Public Shared Function ExecuteScalar(ByVal cmdText As String, Optional ByVal cmdType As CommandType = CommandType.StoredProcedure, Optional ByVal params() As SqlParameter = Nothing) As Object
        Dim cmd As New SqlCommand()
        Dim obj As Object

        Try
            cmd.CommandType = cmdType
            cmd.CommandText = cmdText
            If Not IsNothing(params) Then
                cmd.Parameters.AddRange(params)
            End If
            cmd.Connection = New SqlConnection(ConfigurationManager.ConnectionStrings("LEXSPROD").ConnectionString)
            cmd.Connection.Open()
            obj = cmd.ExecuteScalar
        Catch ex As Exception
            Throw ex
        Finally
            If cmd.Connection.State <> ConnectionState.Closed Then
                cmd.Connection.Close()
            End If
            cmd.Dispose()
            cmd = Nothing
        End Try

        Return obj
    End Function

End Class

File: DailyReportsForManagers/SQLHelper.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Data.SqlClient;

namespace DailyReportsForManagers
{
    class SQLHelper
    {
        public static void ExecuteNonQuery(String cmdText, CommandType cmdType = CommandType.StoredProcedure, List<SqlParameter> paramList = null)
        {

            SqlCommand cmd = new SqlCommand();

            try
            {
                cmd.CommandType = cmdType;
                cmd.CommandText = cmdText;
                if (paramList != null)
                {
                    cmd.Parameters.AddRange(paramList.ToArray());
                }
                cmd.Connection = new SqlConnection("server= lexxware.westus2.cloudapp.azure.com,1433;uid=401Hr3m487%;pwd=&Dogv@S3lfish$;Database=DMS;connect timeout=180");
                cmd.Connection.Open();
                cmd.ExecuteNonQuery();
            }
            catch (DbException e)
            {
                throw e;
            }
            finally
            {
                if (cmd.Connection.State != ConnectionState.Closed)
                {
                    cmd.Connection.Close();
                }
                cmd.Dispose();
                cmd = null;
            }
        }

        public static DataTable GetDataTable(String cmdText, CommandType cmdType = CommandType.StoredProcedure, List<SqlParameter> paramList = null)
        {

            SqlCommand cmd = new SqlCommand();
            DataTable dtTemp = new DataTable();

            try
            {
                cmd.CommandType = cmdType;
                cmd.CommandText = cmdText;
                if (paramList != null)
                {
                    cmd.Parameters.AddRange(paramList.ToArray());
                }
                cmd.Connection = new SqlConnection("server=lexxware.westus2.cloudapp.azure.com,1433;uid=401Hr3m487%;pwd=&Dogv@S3lfish$;Database=DMS;connect timeout=180");
                cmd.Connection.Open();
                SqlDataReader reader = cmd.ExecuteReader();
                dtTemp.Load(reader);
            }
            catch (DbException e)
            {
                throw e;
            }
            finally
            {
                if (cmd.Connection.State != ConnectionState.Closed)
                {
                    cmd.Connection.Close();
                }
                cmd.Dispose();
                cmd = null;
            }

            return dtTemp;
        }

        public static DataSet GetDataSet(String cmdText, CommandType cmdType = CommandType.StoredProcedure, List<SqlParameter> paramList = null)
        {

            SqlCommand cmd = new SqlCommand();
            DataSet dsTemp = new DataSet();

            try
            {
                cmd.CommandType = cmdType;
                cmd.CommandText = cmdText;
                cmd.CommandTimeout = 360; //6min
                if (paramList != null)
                {
                    cmd.Parameters.AddRange(paramList.ToArray());
                }
                cmd.Connection = new SqlConnection("server=lexxware.westus2.cloudapp.azure.com,1433;uid=401Hr3m487%;pwd=&Dogv@S3lfish$;Database=DMS;connect timeout=180");
                SqlDataAdapter da = new SqlDataAdapter(cmd);
                da.Fill(dsTemp);
            }
            catch (DbException e)
            {
                throw e;
            }
            finally
            {
                if (cmd.Connection.State != ConnectionState.Closed)
                {
                    cmd.Connection.Close();
                }
                cmd.Dispose();
                cmd = null;
            }

            return dsTemp;
        }

        public static Object ExecuteScalar(String cmdText, CommandType cmdType = CommandType.StoredProcedure, List<SqlParameter> paramList = null)
        {

            SqlCommand cmd = new SqlCommand();
            Object obj = new Object();

            try
            {
                cmd.CommandType = cmdType;
                cmd.CommandText = cmdText;
                if (paramList != null)
                {
                    cmd.Parameters.AddRange(paramList.ToArray());
                }
                cmd.Connection = new SqlConnection("server=lexxware.westus2.cloudapp.azure.com,1433;uid=401Hr3m487%;pwd=&Dogv@S3lfish$;Database=DMS;connect timeout=180");
                cmd.Connection.Open();
                obj = cmd.ExecuteScalar();
            }
            catch (DbException e)
            {
                throw e;
            }
            finally
            {
                if (cmd.Connection.State != ConnectionState.Closed)
                {
                    cmd.Connection.Close();
                }
                cmd.Dispose();
                cmd = null;
            }
            return obj;
        }
    }
}